### PR TITLE
Added p2p debugger setter

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- using the same logger on all messenger components led to difficulties during debugging
  
## Proposed changes
- added setter for p2p debugger, as it could be optional and set later on from node
- added p2p metrics for full archive network

## Testing procedure
- system test with some nodes on full archive network
- on full archive nodes, `/node/p2pstatus` must have metrics ending with _full_archive, which should reflect the actual state of the network

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/google/gops v0.3.18
 	github.com/gorilla/websocket v1.5.0
 	github.com/mitchellh/mapstructure v1.5.0
-	github.com/multiversx/mx-chain-communication-go v1.0.6
+	github.com/multiversx/mx-chain-communication-go v1.0.7-0.20230918095306-97f7d8d179bf
```
Comment (iulianpascalau) [inline go.mod:16]: proper release after testing

**Review by ssd04 — DISMISSED** [review]

File: go.mod (hunk 1 above)
Comment (sstanculeanu) [reply go.mod]: done

**Review by gabi-vuls — DISMISSED** [review]
Normal allin test: v1.5.13-dev-config-8c685c8e00 -> p2p_debugger_update-9770fd4be1

--- Specific errors ---

block hash does not match 4123
wrong nonce in block 1650
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  0
Nr. of all WARNS:  170
Nr. of new ERRORS:  0
Nr. of new WARNS:  15
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

--- WARNINGS ---

cp-test-02-validator-14 :
Warn: subroundSignature.doSignatureConsensusCheck: fallback validation has been applied minimum number of signatures required = actual number of signatures received =  15

/------/

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — COMMENTED** [review]
Current warnings will be solved with a future PR.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Pull request merged**